### ISSUE TITLE
stat packer-vbox-out.log: no such file or directory

### ISSUE THREAD

**Issue by ck37** [ucberkeley/bce#3]
Trying out the latest version, I get this error when I run "BCE_PROVISION=BCE packer build BCE-14.10-amd64.json" and the script stops executing:

---

1 error(s) occurred:
- Bad source 'packer-vbox-out.log': stat packer-vbox-out.log: no such file or directory
  _____

If I run "touch packer-vbox-out.log" first the script does start building correctly.

**Comment by ryanlovett** [comment]
What version of packer are you using? I cannot reproduce this with 0.7.5.

It would also be best to run "make vbox" instead so that packer.log is created as well.

**Comment by ck37** [comment]
Thanks, this is using packer 0.7.5. Running "make vbox" instead does resolve that issue (did need to remove the --audio parameter from the json per your earlier suggestion) - thanks.

I do seem to be getting the same overall build error from 0.1.4 though, with this key output in packer.log:

(Snippet)
2015/01/09 14:25:15 packer-builder-virtualbox-iso: 2015/01/09 14:25:15 Executing VBoxManage: []string{"startvm", "BCE-2015-spring-preview-2", "--type", "headless"}
2015/01/09 14:25:15 packer-builder-virtualbox-iso: 2015/01/09 14:25:15 stdout: Waiting for VM "BCE-2015-spring-preview-2" to power on...
2015/01/09 14:25:15 packer-builder-virtualbox-iso: 2015/01/09 14:25:15 stderr: VBoxManage: error: No error info
2015/01/09 14:25:15 packer-builder-virtualbox-iso: VBoxManage: error: Details: code NS_ERROR_CALL_FAILED (0x800706be), component ProgressProxy, interface IProgress
2015/01/09 14:25:15 ui: [1;32m==> virtualbox-iso: Waiting 10s for boot...[0m
2015/01/09 14:25:25 ui: [1;32m==> virtualbox-iso: Typing the boot command...[0m
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 Special code '<esc>' found, replacing with: [01 81]
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 Special code <wait> found, will sleep 1 second at this point.
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 Executing VBoxManage: []string{"controlvm", "BCE-2015-spring-preview-2", "keyboardputscancode", "01"}
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 stdout:
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 stderr: VBoxManage: error: Guest not running
2015/01/09 14:25:25 ui error: [1;31m==> virtualbox-iso: Error sending boot command: VBoxManage error: VBoxManage: error: Guest not running[0m
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 Executing VBoxManage: []string{"storageattach", "BCE-2015-spring-preview-2", "--storagectl", "IDE Controller", "--port", "1", "--device", "0", "--medium", "none"}
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 stdout:
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 stderr:
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 Executing VBoxManage: []string{"storageattach", "BCE-2015-spring-preview-2", "--storagectl", "IDE Controller", "--port", "0", "--device", "1", "--medium", "none"}
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 stdout:
2015/01/09 14:25:25 packer-builder-virtualbox-iso: 2015/01/09 14:25:25 stderr:
2015/01/09 14:25:25 ui: [1;32m==> virtualbox-iso: Unregistering and deleting virtual machine...[0m

**Comment by ryanlovett** [comment]
Okay, I'll try the build on a Mac later this afternoon. In other cases
where the build failed and packer was not able to capture relevent
stdout/stderr, I would manually run the commands that packer ran, e.g.

VBoxManage startvm BCE-2015-spring-preview-2 --type headless
VBoxManage controlvm BCE-2015-spring-preview-2 keyboardputscancode 01
VBoxManage storageattach BCE-2015-spring-preview-2 --storagectl "IDE
Controller" --port 1 --device 0 --medium none
VBoxManage storageattach BCE-2015-spring-preview-2 --storagectl "IDE
Controller" --port 0 --device 1 --medium none

In this case it looks like it is failing the startvm command.

Ryan

**Comment by ck37** [comment]
Ok, figured it out on my end at least: had to change --accelerate3d to off in the json file, "vboxmanage" section. For posterity this is on a "late 2013" macbook pro and virtualbox 4.3.20.

**Comment by ck37** [comment]
I still get this error when building ec2, using the latest version of packer (0.8.1):

``` bash
$ make ec2
packer build -only=amazon-ebs \
        -var bce_host_audio=none \
        -var bce_user=ubuntu \
        BCE-2015-summer.json 2>&1 | tee packer-ec2.log 
amazon-ebs output will be in this color.

1 error(s) occurred:

* Bad source 'packer-vbox.log': stat packer-vbox.log: no such file or directory
```

If I touch packer-vbox.log it does resolve the issue, or I can delete the packer-vbox.log section in BCE-2015-summer.json (not sure what it does though!).

**Comment by ck37** [comment]
I found the solution to this, we should add "only": ["virtualbox-iso"] to that provisioner (via https://www.packer.io/docs/templates/provisioners.html) so that it isn't run when building for ec2.

**Comment by ck37** [comment]
Ok, resolved this issue by Commit 375b6c1d46fa96be2287171695eaec3c73d7677e  - closing the issue. I should bring up the audio parameter problem in a separate issue if it's still a problem (need to retry).
